File: code/rachel/Python/lab02.py
x = input('Enter a number: ')
x = int(x)
hundreds_digit = x//100
y = hundreds_digit*100
ones_digit = x%10
tens_digit = x-(y+ones_digit)

def charlie(hundreds_digit):
    if hundreds_digit == 0:
        return('')
    elif hundreds_digit == 1:
        return('one hundred ')
    elif hundreds_digit == 2:
        return('two hundred ')
    elif hundreds_digit == 3:
        return('three hundred ')
    elif hundreds_digit == 4:
        return('four hundred ')
    elif hundreds_digit == 5:
        return('five hundred ')
    elif hundreds_digit == 6:
        return('six hundred ')
    elif hundreds_digit == 7:
        return('seven hundred ')
    elif hundreds_digit == 8:
        return('eight hundred ')
    elif hundreds_digit == 9:
        return('nine hundred ')
def alpha(tens_digit):
    if tens_digit == 10:
        return('ten-')
    elif tens_digit == 20:
        return('twenty-')
    elif tens_digit == 30:
        return('thirty-')
    elif tens_digit == 40:
        return('forty-')
    elif tens_digit == 50:
        return('fifty-')
    elif tens_digit == 60:
        return('sixty-')
    elif tens_digit == 70:
        return('seventy-')
    elif tens_digit == 80:
        return('eighty-')
    elif tens_digit == 90:
        return('ninety-')
def beta(ones_digit):
    if ones_digit == 2:
        return('two-')
    elif ones_digit == 3:
        return('three')
    elif ones_digit == 4:
        return('four')
    elif ones_digit == 5:
        return('five')
    elif ones_digit == 6:
        return('six')
    elif ones_digit == 7:
        return('seven')
    elif ones_digit == 8:
        return('eight')
    elif ones_digit == 9:
        return('nine')

if x == 10:
    print('ten')
elif x == 11:
    print('eleven')
elif x == 12:
    print('twelve')
elif x == 13:
    print('thirteen')
elif x == 14:
    print('fourteen')
elif x == 15:
    print('fifteen')
elif x == 16:
    print('sixteen')
elif x == 17:
    print('seventeen')
elif x == 18:
    print('eighteen')
elif x == 19:
    print('nineteen')
else:
    print(f'{charlie(hundreds_digit)}{alpha(tens_digit)}{beta(ones_digit)}')
    

File: code/josh/javascript/josh_js_lab02/lab02.js
// Lab 2: Number to Phrase
// Convert a given number into its English representation. Handle numbers from 0-999.

let number = prompt("Please enter a number from 0 to 999: ")
number = parseInt(number)
const numberListTo19 = {
    0: "",
    1: "one",
    2: "two",
    3: "three",
    4: "four",
    5: "five",
    6: "six",
    7: "seven",
    8: "eight",
    9: "nine",
    10: "ten",
    11: "eleven",
    12: "twelve",
    13: "thirteen",
    14: "fourteen",
    15: "fifteen",
    16: "sixteen",
    17: "seventeen",
    18: "eighteen",
    19: "nineteen"
}
const numberListForTeensOver100 = {
    11: "eleven",
    12: "twelve",
    13: "thirteen",
    14: "fourteen",
    15: "fifteen",
    16: "sixteen",
    17: "seventeen",
    18: "eighteen",
    19: "nineteen"
}
const numberListByTens = {
    20: "twenty",
    30: "thirty",
    40: "forty",
    50: "fifty",
    60: "sixty",
    70: "seventy",
    80: "eighty",
    90: "ninety"
}
const numberListByTensOver100 = {
    1: "ten",
    2: "twenty",
    3: "thirty",
    4: "forty",
    5: "fifty",
    6: "sixty",
    7: "seventy",
    8: "eighty",
    9: "ninety"
}
const tensPlaceNumberList = {
    0: "",
    2: "twenty-",
    3: "thirty-",
    4: "forty-",
    5: "fifty-",
    6: "sixty-",
    7: "seventy-",
    8: "eighty-",
    9: "ninety-"
}
const hundredsPlaceNumberList = {
    1: "one hundred",
    2: "two hundred",
    3: "three hundred",
    4: "four hundred",
    5: "five hundred",
    6: "six hundred",
    7: "seven hundred",
    8: "eight hundred",
    9: "nine hundred"
}
unitsPlace = number%10
tensPlace = Math.floor(number/10)
hundredsPlace = Math.floor(number/100)
hundredsRemainder = Math.floor(number%100/10)
unitsRemainder = number%100%10
teensRemainder = number%100

// Creates a conditional to print an error message for negative integers out of range.
if number < 0:
    print("Your entry is invalid.  You did not enter a number from 0 to 999.")
// Creates a conditional to print the integer "0" as a word.
elif number == 0:
    print("0 is written out as zero.")
// Creates a conditional to print integers as words if user input is 1 - 19.
elif number <= 19:
    print(f"{number} is written out as {numberListTo19[number]}.")
// Creates a conditional to print integers as words if user input is a multiple of 10, starting at 20
elif number == 20 or number == 30 or number == 40 or number == 50 or number == 60 or number == 70 or number == 80 or number == 90:
    print(f"{number} is written out as {numberListByTens[number]}.")
// Creates a conditional to print integers as words for all other user input less than 100
elif number >= 21 and number <= 99:
    print(f"{number} is written out as {tensPlaceNumberList[tensPlace]}{numberListTo19[unitsPlace]}.")
// Creates a conditional to print integers as words for all numbers from 100 to 999.
elif number >= 100 and number <= 999:
    if number == 100 or number == 200 or number == 300 or number == 400 or number == 500 or number == 600 or number == 700 or number == 800 or number == 900:
        print(f"{number} is written out as {hundredsPlaceNumberList[hundredsPlace]}")
    elif unitsRemainder == 0:
        print(f"{number} is written out as {hundredsPlaceNumberList[hundredsPlace]} {numberListByTensOver100[hundredsRemainder]}")
    elif hundredsRemainder == 1:
        print(f"{number} is written out as {hundredsPlaceNumberList[hundredsPlace]} {numberListForTeensOver100[teensRemainder]}")
    else:
        print(f"{number} is written out as {hundredsPlaceNumberList[hundredsPlace]} {tensPlaceNumberList[hundredsRemainder]}{numberListTo19[unitsRemainder]}.")
// Creates a conditional to print an error message for positive integers out of range.
else:
    print("Your entry is invalid.  You did not enter a number from 0 to 999.")